pkg/scalers/azure_queue.go

```diff
@@ -57,3 +70,22 @@ func GetAzureQueueLength(ctx context.Context, podIdentity string, connectionStri
 
 	return props.ApproximateMessagesCount(), nil
 }
+
+// GetAzureVisibleQueueLength returns the number of visible messages in a queue in int
+func GetAzureVisibleQueueLength(ctx context.Context, podIdentity string, connectionString, queueName string, accountName string, maxCount int32) (int32, error) {
+
+	var err error
+
+	queueURL, err := GetAzureQueueURL(ctx, podIdentity, connectionString, queueName, accountName)
+	if err != nil {
+		return -1, err
+	}
+
+	pmr, err := queueURL.NewMessagesURL().Peek(ctx, maxCount)
```

Comment:
How do you know if it doesn't exceed the max count?

Reply:
Well, the goal of maxCount here is to limit the peak time, as the more message you have to peak, the longer it will be. Max count is the max number of jobs (and so messages) that our scale